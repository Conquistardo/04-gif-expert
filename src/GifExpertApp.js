import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({defaultCategorias = []}) => {
    const [catergorias, setCatergorias] = useState(defaultCategorias);

    return (
        <>
            <h2>GifExperApp</h2>
            <AddCategory setCatergorias = {setCatergorias}/>
            <hr/>
            <ol>
                {
                    catergorias.map(category => (
                         <GifGrid
                                key= {category}
                                categoria= {category}
                            />
                            ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;