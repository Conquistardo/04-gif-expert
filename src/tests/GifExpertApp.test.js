import GifExpertApp from "../GifExpertApp";
import {shallow} from 'enzyme';

describe('Pruebas del coponente GifExpertApp', () => {

    test('el snapshot debe hacer match ', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {

        const categorias = ['One Pounch', 'Ultra instinto'];
        const wrapper = shallow(<GifExpertApp defaultCategorias={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
        
    })
        
    
});