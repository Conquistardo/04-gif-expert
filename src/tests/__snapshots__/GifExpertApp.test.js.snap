// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del coponente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCatergorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Pounch"
      key="One Pounch"
    />
    <GifGrid
      categoria="Ultra instinto"
      key="Ultra instinto"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del coponente GifExpertApp el snapshot debe hacer match  1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCatergorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
